BruTile/Web/HttpTileSource.cs

```diff
@@ -1,81 +1,65 @@
 ﻿// Copyright (c) BruTile developers team. All rights reserved. See License.txt in the project root for license information.
 
+#nullable enable
+
 using System;
 using System.Collections.Generic;
 using System.Net.Http;
 using System.Threading;
 using System.Threading.Tasks;
 using BruTile.Cache;
+using BruTile.Extensions;
 
 namespace BruTile.Web;
 
-public class HttpTileSource : ITileSource, IRequest
+public class HttpTileSource(
+    ITileSchema tileSchema,
+    IUrlBuilder urlBuilder,
+    string? name = null,
+    IPersistentCache<byte[]>? persistentCache = null,
+    Attribution? attribution = null,
+    Action<HttpRequestMessage>? configureHttpRequestMessage = null) : IHttpTileSource, IUrlBuilder
```

Comment:
The configureHttpRequestMessage is one of the core changes of this PR. It allows changes to the HttpRequestMessage used in the http request like the User-Agent and Referer.